### PR TITLE
Deprecate `ContactMap` (allowing rename `ContactFrequency` => `ContactMap`)

### PR DESCRIPTION
**Pull request by dwhswenson**
First step of #82.

Also, note that the correct warning to use here is `FutureWarning`, which will actually show for users (`DeprecationWarning` will not). I learned that from [this SO answer](https://stackoverflow.com/a/55378483/4205735), written by a very brilliant author. I would argue that basic interactive use of Contact Map Explorer in a Jupyter notebook counts as "application", not "library."

### CODE REVIEW

**Comment by dwhswenson** [comment]
Codeclimate complains about:

* contact_map.py being too long (well, it'll be shorter when we remove `ContactMap`! plus, as stated before, I intend to split things out of it in a PR dedicated to that process)
* the changes for deprecating classmethods look like repeated code. While writing the code, I actually thought about writing a decorator to avoid that repetition, but it seemed like overkill for code that's going to be removed in a near-future release.

Marking both as wontfix.

File: contact_map/contact_map.py [hunk 1]
```diff
@@ -7,6 +7,9 @@
 import itertools
 import pickle
 import json
+
+import warnings
+
```
Comment (sroet) [inline contact_map/contact_map.py:12]: Any reason for the excess newlines around this import?

Comment (dwhswenson) [reply contact_map/contact_map.py:12]: reminder that this is temporary and should be removed

**Comment by dwhswenson** [comment]
> * in the contact_map.ipynb there are still a couple references to ContactMap (an import and some text in the `list residue contacts` section

Fixed.

> * The without_atom_slice notebook still references contactMap (as the only bad case), so there might need to be some alteration of the text there (as the example can't get our code to be slower than without atomslice). Maybe try adding a map for a single frame?

I probably should have actually read the text instead of just removing every bit of code that said `ContactMap`.... anyway, I think what's there now illustrates the original point.

**Pull request merged**